pkg/cmd/cli/cmd/newbuild.go

```diff
@@ -186,6 +190,12 @@ func (o *NewBuildOptions) Complete(baseName, commandName string, f *clientcmd.Fa
 	cmdutil.WarnAboutCommaSeparation(o.ErrOut, o.Config.Environment, "--env")
 	cmdutil.WarnAboutCommaSeparation(o.ErrOut, o.Config.BuildEnvironment, "--build-env")
 
+	buildArgs, _, err := cmdutil.ParseEnv(o.BuildArgs, in)
+	if err != nil {
+		return fmt.Errorf(strings.Replace(err.Error(), "environment variable", "build arg", -1))
+	}
+	o.Config.BuildArgs = buildArgs
```

Comment:
i would expect this to be handled in the same place/manner as we handle o.Config.Environment